tests/test_secondary.py

```diff
@@ -0,0 +1,721 @@
+          firmware_fileinfo=factory_firmware_fileinfo,
+          director_public_key=None,
+          partial_verifying=False)
+
+    # Invalid ECU Serial
+    with self.assertRaises(tuf.FormatError):
+      s = secondary.Secondary(
+          full_client_dir=TEMP_CLIENT_DIR_1,
+          director_repo_name=demo.DIRECTOR_REPO_NAME,
+          vin=vin1,
+          ecu_serial=500,
+          ecu_key=secondary_ecu_key,
+          time=clock,
+          timeserver_public_key=key_timeserver_pub,
+          firmware_fileinfo=factory_firmware_fileinfo,
+          director_public_key=None,
+          partial_verifying=False)
+
+    # Invalid ECU Key
+      s = secondary.Secondary(
+          full_client_dir=TEMP_CLIENT_DIR_1,
+          director_repo_name=demo.DIRECTOR_REPO_NAME,
+          vin=vin1,
+          ecu_serial=ecu_serial1,
+          ecu_key={''},
+          time=clock,
+          timeserver_public_key=key_timeserver_pub,
+          firmware_fileinfo=firmware_fileinfo,
+          director_public_key=None,
+          partial_verifying=False)
+
+    # Invalid time:
+    with self.assertRaises(tuf.FormatError):
+      s = secondary.Secondary(
+          full_client_dir=TEMP_CLIENT_DIR_1,
+          director_repo_name=demo.DIRECTOR_REPO_NAME,
+          vin=vin1,
+          ecu_serial=ecu_serial1,
+          ecu_key=secondary_ecu_key,
+          time='potato',
+          timeserver_public_key=key_timeserver_pub,
+          firmware_fileinfo=factory_firmware_fileinfo,
+          director_public_key=key_directortargets_pub,
+          partial_verifying=False)
+
+    # Invalid director_public_key:
+    with self.assertRaises(tuf.FormatError):
+      s = secondary.Secondary(
+          full_client_dir=TEMP_CLIENT_DIR_1,
+          director_repo_name=demo.DIRECTOR_REPO_NAME,
+          vin=vin1,
+          ecu_serial=ecu_serial1,
+          ecu_key=secondary_ecu_key,
+          time=clock,
+          timeserver_public_key=key_timeserver_pub,
+          firmware_fileinfo=factory_firmware_fileinfo,
+          director_public_key={''},
+          partial_verifying=False)
+
+    # Inconsistent arguments, partial_verifying and director_public_key.
+    # partial verification requires a director_public_key argument, as it does
+    # not use the normal trust chain. Providing a director_public_key when not
+    # performing partial verification makes no sense, as the keys to be used
+    # for full verification are determined based on the root metadata file.
+    with self.assertRaises(uptane.Error):
+      s = secondary.Secondary(
+          full_client_dir=TEMP_CLIENT_DIR_1,
+          director_repo_name=demo.DIRECTOR_REPO_NAME,
+          vin=vin1,
+          ecu_serial=ecu_serial1,
+          ecu_key=secondary_ecu_key,
+          time=clock,
+          timeserver_public_key=key_timeserver_pub,
+          firmware_fileinfo=factory_firmware_fileinfo,
+          director_public_key=key_directortargets_pub,
```

Comment:
Instead of using two parameters, pPerhaps make it so that specifying a public director key is understood to mean that partial_verification is in effect.